java/cdk-environment/src/main/java/sleeper/environment/cdk/buildec2/BuildEC2Stack.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.environment.cdk.buildec2;
+
+import sleeper.environment.cdk.config.AppContext;
+import sleeper.environment.cdk.util.MyIpUtil;
+import software.amazon.awscdk.CfnOutput;
+import software.amazon.awscdk.Stack;
+import software.amazon.awscdk.StackProps;
+import software.amazon.awscdk.services.ec2.*;
+import software.constructs.Construct;
+
+import java.security.KeyPair;
+import java.util.Collections;
+import java.util.UUID;
+
+import static sleeper.environment.cdk.config.AppParameters.VPC;
+
+public class BuildEC2Stack extends Stack {
+
+    public BuildEC2Stack(Construct scope, StackProps props, IVpc inheritVpc) {
+        super(scope, props.getStackName(), props);
+        AppContext context = AppContext.of(this);
+        BuildEC2Parameters params = BuildEC2Parameters.from(context);
+        IVpc vpc = context.getOrDefault(VPC, this, inheritVpc);
+
+        SecurityGroup allowSsh = SecurityGroup.Builder.create(this, "AllowSsh")
+                .vpc(vpc)
+                .description("Allow SSH inbound traffic")
+                .allowAllOutbound(true)
+                .build();
+        allowSsh.addIngressRule(Peer.ipv4(MyIpUtil.findMyIp() + "/32"), Port.tcp(22));
+
+        // Create a new SSH key every time the CDK is run.
+        // A UUID is appended to the key name to ensure that the key is deleted and recreated every time the stack is
+        // deployed. This is because AWS does not permit updating key pairs in-place, and the library we're using does
+        // not handle that.
+        KeyPair keyPair = KeyPairUtil.generate();
+        KeyPairUtil.writePrivateToFile(keyPair, props.getStackName() + ".pem");
+        CfnKeyPair key = CfnKeyPair.Builder.create(this, "KeyPair")
+                .keyName(props.getStackName() + "-" + UUID.randomUUID())
+                .publicKeyMaterial(KeyPairUtil.publicBase64(keyPair))
+                .build();
+
+        Instance instance = Instance.Builder.create(this, "EC2")
+                .vpc(vpc)
+                .securityGroup(allowSsh)
+                .machineImage(MachineImage.lookup(LookupMachineImageProps.builder()
+                        .name("ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*")
+                        .owners(Collections.singletonList("099720109477"))
```

Comment:
Worth putting these strings as default BuildEc2 parameters?

Reply:
Some of the other setup relies on the image we're using, like the commands in the user data, the root device name, the user to use for SSH.

Reply:
There may be another version of ubuntu that could be used though so can't see the downsides to putting in a parameter with a default value?

Reply:
Done